server/plugins/configuration/configuration.go

```diff
@@ -0,0 +1,127 @@
+package configuration
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/99designs/httpsignatures-go"
+
+	"github.com/woodpecker-ci/woodpecker/server/model"
+	"github.com/woodpecker-ci/woodpecker/server/remote"
+)
+
+type ConfigAPI struct {
+	endpoint string
+	secret   string
+}
+
+// Same as remote.FileMeta but with json tags and string data
+type config struct {
+	Name string `json:"name"`
+	Data string `json:"data"`
+}
+
+type requestStructure struct {
+	Repo          *model.Repo  `json:"repo"`
+	Build         *model.Build `json:"build"`
+	Configuration []*config    `json:"config"`
+}
+
+func NewAPI(endpoint, secret string) ConfigAPI {
+	return ConfigAPI{endpoint: endpoint, secret: secret}
+}
+
+func (cp *ConfigAPI) IsConfigured() bool {
+	return cp.endpoint != ""
+}
+
+func (cp *ConfigAPI) FetchExternalConfig(ctx context.Context, repo *model.Repo, build *model.Build, currentConfig []*remote.FileMeta) (configData []*remote.FileMeta, useOld bool, err error) {
+	// Create request, needs repo
+	response := struct {
+		Pipelines []config `json:"pipelines"`
```

Comment:
We should name it pipeline or config 😉 

```suggestion
		Pipelines []config `json:"pipelines"`
```

Reply:
So should I change the field name to Configs ? seems to be more in line with what I did so far

Reply:
changed